src/pixelbasher.cpp

```diff
@@ -66,6 +68,7 @@ BMP PixelBasher::compare_bmps(const BMP &original, const BMP &target, bool enabl
         }
     }
     diff.set_data(diff_data);
+    diff.recalculate_masks();
```

Comment:
Splitting `set_data` and `recalculate` into two public member functions is problematic, as it leaves the invariants violated after the first call, and only fixes them with the second.  Better to have `set_data` call `recalculate_masks` internally, so that at the return from `set_data` the invariants hold.  (And if that causes unnecessary recalculations on temporary `BMP` instances for the calls to `set_data` internally in `bmp.cpp`, you could add some private `set_data_internal` or `set_data_unsafe` that skips them and can only be called from within `BMP` code itself, so not getting into trouble with upholding the invariants and the boundaries of the `BMP` code.)